controllers/windowsmachine_controller.go

```diff
@@ -106,7 +106,8 @@ func (r *WindowsMachineReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		},
 		// process delete event
 		DeleteFunc: func(e event.DeleteEvent) bool {
-			return r.isValidMachine(e.Object) && isWindowsMachine(e.Object.GetLabels())
+			// for Windows machines only
+			return isWindowsMachine(e.Object.GetLabels())
```

Comment:
WMCO is able to capture the Deleting event and update the windows-exporter endpoint

```log

2021-09-28T14:11:37.491Z	INFO	controller.windowsmachine	Windows VM has been configured as a worker node	{"ID": "422c28df-5d7a-c518-1b1a-2a3e6af9e66d"}
2021-09-28T14:11:37.494Z	DEBUG	controller-runtime.manager.events	Normal	{"object": {"kind":"Machine","namespace":"openshift-machine-api","name":"e2e-wm-vdmnd","uid":"f74691cd-989b-459a-a267-b5d29506503a","apiVersion":"machine.openshift.io/v1beta1","resourceVersion":"33464"}, "reason": "MachineSetup", "message": "Machine e2e-wm-vdmnd configured successfully"}
2021-09-28T14:11:37.522Z	INFO	metrics	Prometheus configured	{"endpoints": "windows-exporter", "port": 9182, "name": "metrics"}

2021-09-28T14:11:37.522Z	DEBUG	controller.windowsmachine	reconciling	{"windowsmachine": "openshift-machine-api/e2e-wm-vdmnd"}
2021-09-28T14:11:37.526Z	INFO	controller.windowsmachine	machine has current version	{"windowsmachine": "openshift-machine-api/e2e-wm-vdmnd", "version": "2.0.3+087a693"}
2021-09-28T14:11:37.547Z	INFO	metrics	Prometheus configured	{"endpoints": "windows-exporter", "port": 9182, "name": "metrics"}

2021-09-28T14:13:04.440Z	DEBUG	controller.windowsmachine	reconciling	{"windowsmachine": "openshift-machine-api/e2e-wm-vdmnd"}
2021-09-28T14:13:04.440Z	DEBUG	controller.windowsmachine	machine not provisioned	{"windowsmachine": "openshift-machine-api/e2e-wm-vdmnd", "phase": "Deleting"}
2021-09-28T14:13:04.458Z	INFO	metrics	Prometheus configured	{"endpoints": "windows-exporter", "port": 9182, "name": "metrics"}
```